### PR TITLE
Preview

### PR DESCRIPTION
**Pull request by sophieheb**

### CODE REVIEW

File: src/pages/api/preview.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    const previewModeCheck = await axiosGet('/',
+      {
+        headers: {
+          'X-PREVIEW-MODE': 'preview',
+          authorization: `Bearer ${req.query.token}`,
+        },
+      });
+    if (previewModeCheck.previewMode) {
+      res.setPreviewData({});
+      res.status(200).json({ message: "Preview mode turned on! The content you'll see now will include anything marked as saved but not published in storyblok.", allowed: true });
```
Comment (jits) [inline src/pages/api/preview.js:14]: Wording may need some slight adjusting… it currently sounds like published content is not shown when in preview mode – maybe something more along the lines of "You'll now see all the latest content including any new draft content that's not yet published."

File: src/pages/preview-mode.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                      });
+                      setMessage(res.message);
+                      if (res.allowed) {
+                        dispatch(setPreviewMode(true));
+                      } else {
+                        dispatch(setPreviewMode(false));
+                      }
+                    } else {
+                      setMessage('Preview Mode has been turned off.');
+                      dispatch(setPreviewMode(false));
```
Comment (jits) [inline src/pages/preview-mode.js:55]: Does the Next.JS preview mode cookie need clearing too?

File: src/pages/themes/[slug].js [hunk 3]
```diff
@@ -37,10 +42,50 @@ const useStyles = makeStyles((theme) => ({
...
+        authorization: `Bearer ${user.xa}`,
+      };
+
+      axiosGet('themes', {
+        headers,
+      }).then((allThemes) => {
+        setThemes(allThemes.filter((t) => t.slug !== slug));
+        const previewTheme = allThemes.find((t) => t.slug === slug);
+        setTheme(previewTheme);
+        axiosGet('resources', {
```
Comment (jits) [inline src/pages/themes/[slug].js:79]: Need to `return` the promise to chain it with the wrapping promise.

File: src/pages/api/preview.js (hunk 1 above)
Comment (sophieheb) [reply src/pages/api/preview.js:14]: Updated

**Review by jits — APPROVED** [review]
Couple of comments, otherwise LG!

File: src/pages/preview-mode.js (hunk 2 above)
Comment (sophieheb) [reply src/pages/preview-mode.js:55]: Fixed!

**Pull request merged**